subcommand/acl-controller/command.go

```diff
@@ -147,6 +162,101 @@ func (c *Command) Help() string {
 	return ""
 }
 
+// upsertConsulResources creates the necessary resources in Consul if they do not exist.
+// This includes the partition, client role and policy, client and service token auth methods,
+// and the necessary binding rules.
+func (c *Command) upsertConsulResources(consulClient *api.Client, ecsMeta awsutil.ECSTaskMeta) error {
+	account, err := ecsMeta.AccountID()
+	if err != nil {
+		return err
+	}
+	path := strings.Trim(c.flagIAMRolePath, "/")
+	if path != "" {
+		path += "/"
+	}
+	boundArnPattern := fmt.Sprintf("arn:aws:iam::%s:role/%s*", account, path)
+
+	clientAuthMethod := &api.ACLAuthMethod{
+		Name:        "iam-ecs-client-token",
```

Comment:
The auth method name should support a name prefix. I'm going to add that in a follow up PR.

The static names work when partitions are enabled, since we assume one controller per partition. If partitions aren't enabled and there are multiple ECS clusters in a single Consul datacenter, then we can either:

1. Have multiple controllers share an instance of the auth method. Each controller updates the `BoundIAMPrincipalARNs` to trust the IAM roles for its account/cluster. There's a race condition there (multiple controllers updating at the same time), but we could mitigate against that somewhat.
2. OR, alternatively, each controller creates its own instance of the auth method, which requires a unique auth method name. Hence, the name prefix.

(1) may not be so bad now that I've written it out, but it is a hair more complex.